File: lib/py/uapi/internal/types/UFieldDeclaration.py
from typing import Optional

from uapi.internal.types.UTypeDeclaration import UTypeDeclaration


class UFieldDeclaration:
    def __init__(
            self,
            fieldName: str,
            typeDeclaration: UTypeDeclaration,
            optional: bool
    ) -> None:
        self.fieldName = fieldName
        self.typeDeclaration = typeDeclaration
        self.optional = optional

File: lib/ts/uapi/internal/types/UFieldDeclaration.ts
import { UTypeDeclaration } from 'uapi/internal/types/UTypeDeclaration';

export class UFieldDeclaration {
    field_name: string;
    type_declaration: UTypeDeclaration;
    optional: boolean;

    constructor(field_name: string, type_declaration: UTypeDeclaration, optional: boolean) {
        this.field_name = field_name;
        this.type_declaration = type_declaration;
        this.optional = optional;
    }
}
